src/ctds/cursor.c

```diff
@@ -2858,6 +2860,76 @@ static PyGetSetDef Row_getset[] = {
     { NULL,                 NULL,                NULL, NULL,                            NULL }
 };
 
+static const char s_Row_dict_doc[] =
+    "dict()\n"
+    "\n"
+    "Get a dict representing the contents of the row. The keys will be column\n"
+    "names for columns that are named, and integer column numbers for those that\n"
+    "do not.\n"
+    "\n"
+    ":return: A dict representation of the row.\n";
+
+static PyObject* Row_dict(PyObject* self, PyObject* args)
+{
+    struct Row* row = (struct Row*)self;
+    PyObject* dict = PyDict_New();
+
+    if (dict)
+    {
+        PyObject* description = ResultSetDescription_get_object(row->description);
+        if (description)
+        {
+            size_t ncolumns = row->description->ncolumns;
+            size_t ix;
+            assert((Py_ssize_t)ncolumns == PyTuple_GET_SIZE(description));
+            for (ix = 0; ix < ncolumns; ++ix)
+            {
+                PyObject* value = row->values[ix];
+                PyObject* colname = Description_GET_ITEM(PyTuple_GET_ITEM(description, ix), 0);
+                if (0 == PyObject_IsTrue(colname))
```

Comment:
What is this? The equivalent of `if colname:`?

Reply:
Probably more explicit to check/descriptive if `colname` is an empty string. Or am I missing something?

Reply:
Checking if `colname` is an empty string is actually surprisingly more involved, due to pre-Python 3.3 support. Doing the truthy check is supported by all Python versions, and equivalent `if colname:` is arguably what you'd do in Pythonic Python code. I can't imagine writing `if colname == '':`.